crates/matrix-sdk/src/sliding_sync/cache.rs

```diff
@@ -7,11 +7,16 @@
 use std::collections::BTreeMap;
 
 use matrix_sdk_base::{StateStore, StoreError};
+use ruma::UserId;
 use tracing::{trace, warn};
 
 use super::{FrozenSlidingSync, FrozenSlidingSyncList, SlidingSync, SlidingSyncList};
 use crate::{sliding_sync::SlidingSyncListCachePolicy, Client, Result};
 
+pub(super) fn format_base_storage_key_for_sliding_sync(loop_id: &str, user_id: &UserId) -> String {
+    format!("{}::{}", loop_id, user_id.to_string())
```

Comment:
I would prefix with `matrix_sdk::sliding_sync::` or something like that to avoid possible naming conflicts.

Reply:
Done; I've also rejiggered how the other keys were computed. As a note: that means we're changing them, and this will invalidate the cache for all the current users.